ink-wrapper-types/src/drink/client.rs

```diff
@@ -0,0 +1,164 @@
+        call: InstantiateCall<T>,
+    ) -> Result<ContractInstantiateResult<MinimalRuntimeAccount>, Error> {
+        let actor = self.get_actor();
+        let gas_limit = self.get_gas_limit();
+
+        let instantiate_contract_result = self.sandbox().instantiate_contract(
+            call.code_hash.to_vec(),
+            call.value,
+            call.data,
+            call.salt,
+            actor,
+            gas_limit,
+            None,
+        );
+
+        let contract_address = match &instantiate_contract_result.result {
+            Ok(exec_result) if exec_result.result.did_revert() => Err(Error::DeploymentReverted),
+            Err(err) => Err(Error::DeploymentFailed(*err)),
+            Ok(exec_result) => Ok(exec_result.account_id.clone()),
+        }?;
+
+        let events = extract_events(&instantiate_contract_result.events);
+
+        Ok(ContractInstantiateResult {
+            gas_consumed: instantiate_contract_result.gas_consumed,
+            gas_required: instantiate_contract_result.gas_required,
+            result: contract_address,
+            events,
+            reverted: false,
+        })
+    }
+
+    fn execute<T: scale::Decode + Send + std::fmt::Debug>(
+        &mut self,
+        call: ExecCall<T>,
+    ) -> Result<ContractExecResult<T>, Error> {
+        let actor = self.get_actor();
+        let gas_limit = self.get_gas_limit();
+        let contract_address = (*AsRef::<[u8; 32]>::as_ref(&call.account_id)).into();
+
+        call_contract(
+            actor,
+            gas_limit,
+            self.sandbox(),
+            contract_address,
+            call.value,
+            call.data,
+        )
+    }
+
+    fn query<T: scale::Decode + Send + std::fmt::Debug>(
+        &mut self,
+        call: ReadCall<T>,
+    ) -> Result<ContractReadResult<T>, Error> {
+        let actor = self.get_actor();
+        let gas_limit = self.get_gas_limit();
+        let contract_address = (*AsRef::<[u8; 32]>::as_ref(&call.account_id)).into();
+
+        self.sandbox().dry_run(|sandbox| {
+            call_contract(actor, gas_limit, sandbox, contract_address, 0, call.data)
+        })
+    }
+}
+
+fn call_contract<T: scale::Decode + Send + std::fmt::Debug>(
+    actor: MinimalRuntimeAccount,
+    gas_limit: Weight,
+    sandbox: &mut drink::Sandbox<MinimalRuntime>,
+    address: MinimalRuntimeAccount,
+    value: u128,
+    data: Vec<u8>,
+) -> Result<ContractResult<T>, Error> {
+    // Reset events to make sure we don't have any events from previous calls.
+    sandbox.reset_events();
+
+    let result = sandbox.call_contract(
+        address,
+        value,
+        data,
+        actor,
+        gas_limit,
+        None,
+        pallet_contracts::Determinism::Enforced,
+    );
+
+    let message_result: T = match &result.result {
+        // NOTE to reviewers: we're not checking if the contract call reverted here.
+        // We will most probably return an error from the contract call anyway.
```

Comment:
Request: either delete the comment or change it to:
1. Just be a NOTE, reviewers will be done with this at some point :)
2. Explain _why_ we will most probably return an error, so that this knowledge does not need to be reconstructed by the reader.

Reply:
I've removed the comment entirely. I couldn't figure out what I meant to say anymore.